lib/Display/BitmapRegion.cpp

```diff
@@ -38,3 +60,48 @@ void BitmapRegion::render(GxEPD2_GFX* display) {
         backgroundColor);
   }
 }
+
+Rectangle BitmapRegion::getBoundingBox() {
+  return {
+    .x = std::min(this->boundingBox.x, this->previousBound.x),
+    .y = std::min(this->boundingBox.y, this->previousBound.y),
+    .w = std::max(this->boundingBox.w, this->previousBound.w),
+    .h = std::max(this->boundingBox.h, this->previousBound.h)
+  };
+}
+
+bool BitmapRegion::getBitmapDimensions(const String& bitmapName, uint16_t *width, uint16_t *height){
+  String metadata_filename = bitmapName.c_str();
+  Serial.println("AAAAAAAAAAAA");
+  
+  metadata_filename.replace("/b/", "/m/");
+  Serial.println(F("getting dimensions of bitmap:"));
+    Serial.println(bitmapName);
+    Serial.println(metadata_filename);
+  if (!SPIFFS.exists(bitmapName) || !SPIFFS.exists(metadata_filename)) {
+    Serial.println(F("WARN - tried to get dimensions from bitmap that doesn't exist: "));
+    Serial.println(bitmapName);
+    Serial.println(metadata_filename);
+    return false;
+  }
+
+  File metadata_file = SPIFFS.open(metadata_filename, "r");
+
+  DynamicJsonDocument jsonBuffer(2048);
```

Comment:
We probably only need ~100 bytes here.  `StaticJsonDocument` uses stack memory rather than heap, which is generally preferable.

```suggestion
  StaticJsonDocument<100> jsonBuffer;
```